lib/index.js

```diff
@@ -6,111 +6,125 @@ const { getDungeonExpWeight } = require("./dungeonExpWeight");
+        let skillXP;
+        if (!profile.experience_skill_mining) {
+            // skill API is off
+            const player = await getPlayer(uuid, this.apiKey);
+            const { achievements } = player;
+            username = player.displayName;
 
-    console.log(skillLevels, skillXP, cataCompl, mCataCompl, cataXP, slayerXP);
-    const weight = getWeightRaw(skillLevels, skillXP, cataCompl, mCataCompl, cataXP, slayerXP);
+            skillLevels = Object.values(skillNames).map(i => achievements?.[i] ?? 0);
+            skillXP = skillLevels.map(getXpFromLevel);
+        } else {
+            skillXP = Object.keys(skillNames).map(i => profile[i]);
+            skillLevels = skillXP.map(getLevelFromXP);
+        }
 
-    if (returnUsername) {
-        if (!username)
-            username = await fetchUsername(uuid);
-        return { username, uuid, ...weight };
-    } else return { uuid, ...weight };
-};
+        const weight = LilyWeight.getWeightRaw(skillLevels, skillXP, cataCompl, mCataCompl, cataXP, slayerXP);
 
-module.exports = apiKey => {
-    return {
-        /**
-         * Gets the player's weight.
-         * @param {string} player - Either a Minecraft username or a Minecraft UUID (can be with dashes).
-         * @param {boolean} [returnUsername=false] - Should the function return the player's username.
-         * @returns {Promise<{total: number, slayer: number, skill: {overflow: number, base: number}, uuid: string, username?: string, catacombs: {completion: {base: number, master: number}, experience: number}}>}
-         */
-        getWeight: async (player, returnUsername = false) =>
-            getWeight(player.length > 16 ? player : await getUUID(player), apiKey, returnUsername),
+        if (returnUsername) {
+            if (!username)
+                username = await fetchUsername(uuid);
+            return { username, uuid, ...weight };
+        } else return { uuid, ...weight };
+    }
 
-        /**
-         * Gets the player's weight.
-         * @param {string} uuid - Player's Minecraft UUID, can be with dashes.
-         * @param {boolean} [returnUsername=false] - Should the function return the player's username.
-         * @returns {Promise<{total: number, slayer: number, skill: {overflow: number, base: number}, uuid: string, username?: string, catacombs: {completion: {base: number, master: number}, experience: number}}>}
-         */
-        getWeightFromUUID: async (uuid, returnUsername = false) =>
-            getWeight(uuid, apiKey, returnUsername),
+    /**
+     * Gets the player's weight.
+     * @param {string} username - Player's Minecraft username.
+     * @param {boolean} [returnUsername=false] - Should the function return the player's username.
+     * @returns {Promise<PlayerWeights>} the weights calculated for the player.
+     */
+    async getWeightFromUsername(username, returnUsername = false) {
+        return this.getWeight(await getUUID(username), returnUsername);
```

Comment:
This is technically my mistake, but this should use `this.getWeightFromUUID` because we are always providing an UUID.